sound/soc/intel/boards/sof_cirrus_common.c

```diff
@@ -168,11 +168,16 @@ static int cs35l41_compute_codec_conf(void)
 			continue;
 		}
 		physdev = get_device(acpi_get_first_physical_node(adev));
+		acpi_dev_put(adev);
+		if (!physdev) {
+			pr_devel("Cannot find physical node for HID %s UID %u (%s)\n", CS35L41_HID,
+					uid, cs35l41_name_prefixes[uid]);
+			continue;
```

Comment:
return 0?

With the continue, you could hypothetically fail on the last device and still report a non-zero set of codecs.

Reply:
yes but asoc properly handles this failure if its trying to register extra nodes, do we really want to stop registering the rest of the nodes?

Reply:
fair enough, if we know the firmware says its there but we are having trouble finding it that is a pretty clear